Mathlib/Tactic/Linter/UnusedTactic.lean

```diff
@@ -66,31 +66,46 @@ namespace UnusedTactic
 /-- The monad for collecting the ranges of the syntaxes that do not modify any goal. -/
 abbrev M := StateRefT (HashMap String.Range Syntax) IO
 
-/-- `Parser`s allowed to not change the tactic state. -/
-def allowed : HashSet SyntaxNodeKind:= HashSet.empty
-  |>.insert `Mathlib.Tactic.Says.says
-  |>.insert `Batteries.Tactic.«tacticOn_goal-_=>_»
-  -- attempt to speed up, by ignoring more tactics
-  |>.insert `by
-  |>.insert `null
-  |>.insert `«]»
-  |>.insert ``Lean.Parser.Term.byTactic
-  |>.insert ``Lean.Parser.Tactic.tacticSeq
-  |>.insert ``Lean.Parser.Tactic.tacticSeq1Indented
-  |>.insert ``Lean.Parser.Tactic.tacticTry_
-
-  -- the following `SyntaxNodeKind`s play a role in silencing `test`s
-  |>.insert ``Lean.Parser.Tactic.guardHyp
-  |>.insert ``Lean.Parser.Tactic.guardTarget
-  |>.insert ``Lean.Parser.Tactic.failIfSuccess
-  |>.insert `Mathlib.Tactic.successIfFailWithMsg
-  |>.insert `Mathlib.Tactic.failIfNoProgress
-  |>.insert `Mathlib.Tactic.ExtractGoal.extractGoal
-  |>.insert `Mathlib.Tactic.Propose.propose'
-  |>.insert `Lean.Parser.Tactic.traceState
-  |>.insert `Mathlib.Tactic.tacticMatch_target_
-  |>.insert `change?
-  |>.insert `«tactic#adaptation_note_»
+/-- `Parser`s allowed to not change the tactic state.
+This can be increased dynamically, using `#allow_unused_tactic`.
+-/
+initialize allowedRef : IO.Ref (HashSet SyntaxNodeKind) ←
+  IO.mkRef <| HashSet.empty
+    |>.insert `Mathlib.Tactic.Says.says
+    |>.insert `Batteries.Tactic.«tacticOn_goal-_=>_»
+    -- attempt to speed up, by ignoring more tactics
+    |>.insert `by
+    |>.insert `null
+    |>.insert `«]»
+    |>.insert ``Lean.Parser.Term.byTactic
+    |>.insert ``Lean.Parser.Tactic.tacticSeq
+    |>.insert ``Lean.Parser.Tactic.tacticSeq1Indented
+    |>.insert ``Lean.Parser.Tactic.tacticTry_
+    -- the following `SyntaxNodeKind`s play a role in silencing `test`s
+    |>.insert ``Lean.Parser.Tactic.guardHyp
+    |>.insert ``Lean.Parser.Tactic.guardTarget
+    |>.insert ``Lean.Parser.Tactic.failIfSuccess
+    |>.insert `Mathlib.Tactic.successIfFailWithMsg
+    |>.insert `Mathlib.Tactic.failIfNoProgress
+    |>.insert `Mathlib.Tactic.ExtractGoal.extractGoal
+    |>.insert `Mathlib.Tactic.Propose.propose'
+    |>.insert `Lean.Parser.Tactic.traceState
+    |>.insert `Mathlib.Tactic.tacticMatch_target_
+    |>.insert `change?
+    |>.insert `«tactic#adaptation_note_»
+
+/-- `#allow_unused_tactic` takes an input a space-separated list of identifiers and extends
+the tactics that are allowed by the unused tactic linter to not modify goals using the
+input identifiers.
+For instance, you can allow `done` and `skip` to not emit warning using
```

Comment:
Two wording suggestions:
- The main sentence is understandable, but hard to parse - can this be simplified? Here's an attempt:
`#allow_unused_tactic` takes as input a space-separated list of identifiers. These identifier also are allowed by the unused tactic linter: even if these tactics do not modify goals, there will be no warning emitted. Note: for this to work, these identifiers should be the `SyntaxNodeKind` of each tactic.
- What do you think about this rephrasing for the example? "For instance, you can allow the `done` and `skip` tactics using..."

Reply:
Thanks, I incorporated your suggestions, with some very minor adaptations!